joblib/test/test_memory.py

```diff
@@ -229,8 +229,8 @@ def name_collision(x):
         a(1)
         b(1)
 
-        yield assert_equal, len(w), 1
-        yield assert_true, "collision" in str(w[-1].message)
+        assert len(w) == 1
```

Comment:
Makes sense but if I had not read your comment, till now I would have been a believer of the fact that my code block should only raise one warning and two warnings are not expected because they were not desired ( because I wrote `assert len(w) == 1` ) in the first place. Anyhow tests are for a developer to verify the code's functioning and one may design tests in a way which better communicates the health of the corresponding source.

I will try to find the pytest equivalent of yield based tests, if not - I will restore this back but this setup teardown will be replaced y a fixture. I guess the errors should not exist at the end.